Source/WebCore/css/SelectorChecker.cpp

```diff
@@ -281,7 +281,7 @@ SelectorChecker::MatchResult SelectorChecker::matchRecursively(CheckingContext&
         return MatchResult::fails(Match::SelectorFailsLocally);
 
     if (context.selector->match() == CSSSelector::Match::PseudoElement) {
-        if (context.selector->isUserAgentPartPseudoElement()) {
+        if (context.selector->isUserAgentPartPseudoElement() || context.selector->pseudoElement() == CSSSelector::PseudoElement::WebKitUnknown) {
```

Comment:
`PseudoElement::WebKitUnknown` should never match anything, we should fast track all the code below

Reply:
I would prefer making that as a separate change given the risk pointed out below. Unless you're sure there cannot be any web observable impact.

Reply:
I'm pretty sure in this case that there's no web impact. With the current patch, selector matching will fail anyway, but it will go through the work of navigating shadow roots and stuff when it doesn't really need to. We can just fast track this and fail the matching early.